### PR TITLE
cleanup `followAlias`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/mixin.ts [hunk 1]
```diff
@@ -303,18 +303,18 @@ function mergeMixins(
   }
 }
 
-function followAlias(
+function followMethodAlias(
   obj: object,
-  desc: any,
+  _desc: Alias,
```
Comment (bekzod) [inline packages/ember-metal/lib/mixin.ts]: avoid modifying argument

File: packages/ember-metal/lib/mixin.ts [hunk 2]
```diff
@@ -303,18 +303,18 @@ function mergeMixins(
...
-  let value;
+  let altKey = _desc.methodName;
   let possibleDesc;
-  if (descs[altKey] || values[altKey]) {
-    value = values[altKey];
-    desc = descs[altKey];
+  let desc = descs[altKey];
+  let value = values[altKey];
+
+  if (desc !== undefined || value !== undefined) {
```
Comment (bekzod) [inline packages/ember-metal/lib/mixin.ts]: avoid boolean coercion

**Comment by locks** [comment]
@bekzod can you rebase?

**Comment by bekzod** [comment]
rebased @locks

**Review by rwjblue — APPROVED** [review]

**Pull request merged**